packages/internal/src/build/babel/api.ts

```diff
@@ -0,0 +1,89 @@
+import fs from 'fs'
+import path from 'path'
+
+import type { TransformOptions, PluginItem } from '@babel/core'
+
+import { getPaths } from '../../paths'
+
+import { registerBabel } from './common'
+
+export const getApiSideBabelPlugins = () => {
+  const rwjsPaths = getPaths()
+  // Plugin shape: [ ["Target", "Options", "name"] ],
+  // a custom "name" is supplied so that user's do not accidently overwrite
+  // Redwood's own plugins.
+  const plugins: TransformOptions['plugins'] = [
+    ['@babel/plugin-transform-typescript', undefined, 'rwjs-babel-typescript'],
+    [
+      require('@redwoodjs/core/dist/babelPlugins/babel-plugin-redwood-src-alias')
+        .default,
+      {
+        srcAbsPath: rwjsPaths.api.src,
+      },
+      'rwjs-babel-src-alias',
+    ],
+    [
+      require('@redwoodjs/core/dist/babelPlugins/babel-plugin-redwood-directory-named-import')
+        .default,
+      undefined,
+      'rwjs-babel-directory-named-modules',
+    ],
+    [
+      'babel-plugin-auto-import',
+      {
+        declarations: [
+          {
+            // import gql from 'graphql-tag'
+            default: 'gql',
+            path: 'graphql-tag',
+          },
+          {
+            // import { context } from '@redwoodjs/api'
+            members: ['context'],
+            path: '@redwoodjs/api',
+          },
+        ],
+      },
+      'rwjs-babel-auto-import',
+    ],
+    // FIXME: Babel plugin GraphQL tag doesn't seem to be working.
+    ['babel-plugin-graphql-tag', undefined, 'rwjs-babel-graphql-tag'],
+    [
+      require('@redwoodjs/core/dist/babelPlugins/babel-plugin-redwood-import-dir')
+        .default,
+      undefined,
+      'rwjs-babel-glob-import-dir',
+    ],
+  ].filter(Boolean)
+
+  return plugins
+}
+
+export const getApiSideBabelConfigPath = () => {
+  const p = path.join(getPaths().api.base, 'babel.config.js')
+  if (fs.existsSync(p)) {
+    return p
+  } else {
+    return undefined
```

Comment:
Ah yes... sorry I should change this to false. I had it as undefined because I was using extends.

Reply:
I read the docs, it might not actually be searching for `.babelrc.js`, but I would still prefer it to be turned off completely.

Reply:
Peter, so using false _is_ different from using undefined. Using false just makes it fail... my suspicion babel register behaves differently to babel transform.

Reply:
What I'm not loving about this PR is that it feels like it's introducing more uncertainty in a process that I would love to be solid. This is the outcome I'm trying to achieve:

So this is what I want:

1. I want people to delete `api/.babelrc.js`, or to convert it to `api/babel.config.js` and remove the "extends" part.
2. I want this to be a breaking change if you have custom babel transformations. The reason I want it to be a breaking change is because I want to discourage people from adding their own babel transformations because I would like to move our own transformations to a faster toolchain and would not like babel to make that process difficult.

Can we work around making that a realistic goal in this PR instead of muddying the water? And is it something that we can have tests for?

1. Test that `<side>/.babelrc.js` is _not_ imported
2. Test that `<side>/babel.config.json` can be imported if it's available.

Reply:
^^ @thedavidprice Does this reasoning sound solid to you?

Reply:
@peterp The items 1 and 2 above regarding your plan to deprecate and _why_ we should deprecate are SuperImportant™. I'm 100% in agreement with 2 and moving forward with implementation on 1. (But not in this PR, see "next steps" below.)

Instead of this information being buried here, however, what would be really helpful is having this plan communicated in an Issue for visibility and alignment. I'm very interested in the move to a "faster toolchain" (tell me more!), I just don't know what that means, when it would happen, what it will require, etc. 

I think I understand what you mean by "introducing more uncertainty". In this PR, we are deprecating bablerc, but because it won't throw an app error anymore you are concerned it will be confusing, correct? I could go either way. Regardless, I think we can resolve this with 1) good Release Note communication and 2) a two-step process to full deprecation (e.g. move forward with v0.36 and throw if present in v0.37).

## Next Steps
We've been 2 weeks on this bug as a critical-path blocker to releasing v0.36. The code here is an improvement and, most importantly, resolves the release-blocking bugs.

- I'll approve and merge, then cut the v0.36 release branch
- I'll open an Issue with these next steps for v0.37
- PP, I'll leave next steps regarding communicating a plan for item 2 above (babel + faster toolchain) if applicable

Reply:
Just to close the loop on this --  we spoke offline.

The PR where I removed the `.babelrc.js` files (#3187) includes a deprecation notice for the release notes. I'll update the follow on PR #3245 with a bit more information about next steps.

Reply:
And the current Release Notes draft is very clear about the Breaking change + Code Mods regarding `I want people to delete api/.babelrc.js, or to convert it to api/babel.config.js and remove the "extends" part.` — says this explicitly.

Sound good?